lib/bn.js

```diff
@@ -494,6 +494,100 @@ BN.prototype.neg = function neg() {
   return r;
 };
 
+
+// Or `num` with `this` in-place
+BN.prototype.ior = function ior(num) {
+  this.sign = this.sign || num.sign;
+
+  // a.length > b.length
+  var a;
+  var b;
+  if (this.length > num.length) {
+    a = this;
+    b = num;
+  } else {
+    a = num;
+    b = this;
+  }
+  for (var i = 0; i < b.length; i++) {
+    this.words[i] = this.words[i] | num.words[i];
```

Comment:
While this loop does not produce out of bounds reads - it does not use higher limbs for `i` between `b.length` and `a.length`